drivers/gpio/gpio_silabs_siwx91x.c

```diff
@@ -158,6 +136,67 @@ static int gpio_siwx91x_pin_configure(const struct device *dev, gpio_pin_t pin,
 
 	sl_si91x_gpio_set_pin_direction(cfg->hal_port, pin, (flags & GPIO_OUTPUT) ? 0 : 1);
 
+	for (i = 0; i < data->next_pin; i++) {
+		if (data->pin_config_info[i].pin == pin) {
+			cur_cfg_pin = i;
+			break;
+		}
+	}
+
+	if (i == data->next_pin) {
+		cur_cfg_pin = data->next_pin;
+		data->next_pin++;
+	}
+
+	if (cur_cfg_pin < __builtin_popcount(cfg->common.port_pin_mask)) {
+		data->pin_config_info[cur_cfg_pin].port_dev = dev;
+		data->pin_config_info[cur_cfg_pin].pin = pin;
+		data->pin_config_info[cur_cfg_pin].flags = flags;
+	} else {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int gpio_siwx91x_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct gpio_siwx91x_common_config *pcfg = dev->config;
+	struct gpio_siwx91x_common_data *pdata = dev->data;
```

Comment:
Does the `p` prefix for these variable names stand for "port", or are you using Hungarian notation? In both cases I think you should just drop it.